solidity/contracts/converter/types/liquidity-pool-v1/LiquidityPoolV1Converter.sol

```diff
@@ -238,414 +243,426 @@ contract LiquidityPoolV1Converter is LiquidityPoolConverter {
+	 *
+	 * @param _amount  amount to liquidate (in the pool token)
+	 */
+	function liquidate(uint256 _amount) external protected {
+		require(_amount > 0, "ERR_ZERO_AMOUNT");
+
+		uint256 totalSupply = ISmartToken(anchor).totalSupply();
+		ISmartToken(anchor).destroy(msg.sender, _amount);
+
+		uint256[] memory reserveMinReturnAmounts = new uint256[](reserveTokens.length);
+		for (uint256 i = 0; i < reserveMinReturnAmounts.length; i++) reserveMinReturnAmounts[i] = 1;
+
+		removeLiquidityFromPool(reserveTokens, reserveMinReturnAmounts, totalSupply, _amount);
+	}
+
+	/**
+	 * @dev verifies that a given array of tokens is identical to the converter's array of reserve tokens
+	 * we take this input in order to allow specifying the corresponding reserve amounts in any order
+	 *
+	 * @param _reserveTokens   array of reserve tokens
+	 * @param _reserveAmounts  array of reserve amounts
+	 * @param _amount          token amount
+	 */
+	function verifyLiquidityInput(
+		IERC20Token[] memory _reserveTokens,
+		uint256[] memory _reserveAmounts,
+		uint256 _amount
+	) private view {
+		uint256 i;
+		uint256 j;
+
+		uint256 length = reserveTokens.length;
+		require(length == _reserveTokens.length, "ERR_INVALID_RESERVE");
+		require(length == _reserveAmounts.length, "ERR_INVALID_AMOUNT");
+
+		for (i = 0; i < length; i++) {
+			// verify that every input reserve token is included in the reserve tokens
+			require(reserves[_reserveTokens[i]].isSet, "ERR_INVALID_RESERVE");
+			for (j = 0; j < length; j++) {
+				if (reserveTokens[i] == _reserveTokens[j]) break;
+			}
+			// verify that every reserve token is included in the input reserve tokens
+			require(j < length, "ERR_INVALID_RESERVE");
+			// verify that every input reserve token amount is larger than zero
+			require(_reserveAmounts[i] > 0, "ERR_INVALID_AMOUNT");
+		}
+
+		// verify that the input token amount is larger than zero
+		require(_amount > 0, "ERR_ZERO_AMOUNT");
+	}
+
+	/**
+	 * @dev adds liquidity (reserve) to the pool
+	 *
+	 * @param _reserveTokens   address of each reserve token
+	 * @param _reserveAmounts  amount of each reserve token
+	 * @param _totalSupply     token total supply
+	 */
+	function addLiquidityToPool(
+		IERC20Token[] memory _reserveTokens,
+		uint256[] memory _reserveAmounts,
+		uint256 _totalSupply
+	) private returns (uint256) {
+		if (_totalSupply == 0) return addLiquidityToEmptyPool(_reserveTokens, _reserveAmounts);
+		return addLiquidityToNonEmptyPool(_reserveTokens, _reserveAmounts, _totalSupply);
+	}
+
+	/**
+	 * @dev adds liquidity (reserve) to the pool when it's empty
+	 *
+	 * @param _reserveTokens   address of each reserve token
+	 * @param _reserveAmounts  amount of each reserve token
+	 */
+	function addLiquidityToEmptyPool(IERC20Token[] memory _reserveTokens, uint256[] memory _reserveAmounts) private returns (uint256) {
+		// calculate the geometric-mean of the reserve amounts approved by the user
+		require(isOwner(), "ERR_INIT_LIQUIDITY_ONLY_OWNER");

```

Comment:
Use modifier? So that we can remove the redundant error message

Reply:
this is to avoid on contract to hit the size limit

Reply:
considering there is only 1 line code reverted to `onlyOwner` modifier